src/main/java/ro/unibuc/hello/controller/UserController.java

```diff
@@ -12,15 +12,15 @@
 import ro.unibuc.hello.data.UserRepository;
 import ro.unibuc.hello.dto.AuthenticationData;
 import ro.unibuc.hello.dto.UserProfile;
-import ro.unibuc.hello.dto.UserSignUp;
+import ro.unibuc.hello.dto.UserRegister;
 
 @Controller
```

Comment:
I think we should go for @RestController, because we serialize java objects to json representation, we don't return a view

Reply:
I don't think it is related to what we are returning (objects are already serialized as JSONs). (Baeldung: `Spring 4.0 introduced the @RestController annotation in order to simplify the creation of RESTful web services. It's a convenient annotation that combines @Controller and @ResponseBody`)

But good idea anyway